subjects/devops/play-with-containers/README.md

```diff
@@ -0,0 +1,88 @@
+## play-with-containers
+
+![Microservices](pictures/Microservices.png)
+
+### Objectives
+
+This project aims to discover the container concepts and tools, and practice these tools by creating a microservices architecture with docker and docker-compose. Additionally, as a cloud and DevOps student, you will gain insight into key containers and docker concepts as follows:
+
+- `Containers`, is a unit of software that packages code and its dependencies so the application runs quickly and reliably across computing environments.
+- `Docker`, is a set of "platform as a service" products that use OS-level virtualization to deliver software in packages called containers.
+- `Dockerfile`, is a text document that contains all the commands a user could call on the command line to assemble an image.
+- `Docker Images`, is a read-only template that contains a set of instructions for creating a container that can run on the Docker platform.
+- `Docker Networks`, it's enables a user to link a Docker container to as many networks as he/she requires.
+- `Docker Volumes`, Volumes are the preferred mechanism for persisting data generated by and used by Docker containers.
+- `Docker Compose`, Compose is a tool for defining and running multi-container Docker applications. With Compose, you use a YAML file to configure your application's services.
+
+### Tips
+
+- Spend time on the theory before rushing into the practice.
+- Read the official documentation of docker.
+
+> Any lack of understanding of the concepts of this project may affect the difficulty of future projects, Take your time to understand all concepts.
+
+> Be curious and never stop searching!
+
+> Each operation in Dockerfile is a layer in the image, You must design it appropriately to avoid duplicate or useless layers in the image.
+
+> It is not recommended to use 'latest' in your Dockerfile, instead of that you can specify the version in the tag,
+
+### Instructions
+
+You have to install docker in your Linux virtual machine.
+
+> You will need it in the audit.
+
+You have to implement this architecture:
+![architecture](pictures/architecture.png)

```

Comment:
what i didn't understand here is why there two ways of communication between the gateway and the billing app and how the gateway should know which way of commination to use